include/networkit/distance/DynPrunedLandmarkLabeling.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef NETWORKIT_DISTANCE_DYN_PRUNED_LANDMARK_LABELING_HPP_
+#define NETWORKIT_DISTANCE_DYN_PRUNED_LANDMARK_LABELING_HPP_
+
+#include <stdexcept>
+#include <vector>
+
+#include <networkit/base/DynAlgorithm.hpp>
+#include <networkit/distance/PrunedLandmarkLabeling.hpp>
+#include <networkit/dynamics/GraphEvent.hpp>
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+
+/**
+ * @ingroup distance
+ * Dynamic pruned landmark labeling.
+ */
+class DynPrunedLandmarkLabeling final : public PrunedLandmarkLabeling, public DynAlgorithm {
+
+public:
+    /**
+     * Dynamic Pruned Landmark Labeling algorithm based on the paper "Fully Dynamic 2-Hop Cover
+     * Labeling " from D'Angelo et al., ACM JEA 2019. The algorithm computes distance labels by
+     * performing pruned breadth-first searches from each vertex. Distance labels can be updated
+     * efficiently after edge insertions.
+     * @note this algorithm only works for unweighted graphs and only supports edge insertions.
+     *
+     * @param G The input graph.
+     */
+    DynPrunedLandmarkLabeling(const Graph &G) : PrunedLandmarkLabeling(G) {}
+
+    ~DynPrunedLandmarkLabeling() override = default;
+
+    /**
+     * Updates the distance labels after an edge insertion on the graph.
+     * @note supports only edge insertions.
+     *
+     * @param e The edge insertion.
+     */
+    void update(GraphEvent e) override;
+
+    /**
+     * Not implemented. The algorithm does not support batch updates.
```

Comment:
I think it would be nice if `updateBatch` would just call `update` for each event instead of raising an error. Even though there is no special batch update algorithm, users can still use batches for convenience. Most of our other dynamic algorithms do not support both single and batch updates either, but they all forward the call to the other function.

Reply:
Unfortunately that would not work. The algorithm does not update the distances properly if one adds multiple edges at a time and then calls `update` for each of them.